### PR TITLE
Redis Utils

### PR DESCRIPTION
**Pull request by willronchetti**
- Creates 2 new files `redis_utils` and `redis_tools`, the former contains low level APIs for interacting with Redis as a basic key/value store and the latter contains a higher level interface for managing sessions with Redis
- Remove unused `pytest.ini` file
- Update/add dependencies for Redis
- Some basic tests

### CODE REVIEW

File: dcicutils/redis_utils.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+              sub classes later on to isolate in a reasonable manner, but for now all here
+    """
+
+    def __init__(self, redis_handle):
+        self.redis = redis_handle
+
+    @staticmethod
+    def _encode_value(value):
+        """ Function used to preprocess all values that are strings and encode them as utf-8 bytes """
+        return value.encode('utf-8') if (value and isinstance(value, str)) else value
```
Comment (netsettler) [inline dcicutils/redis_utils.py:41]: So this is not needed for other types? Or those other types are not given? Something about the expected args would be helpful in the doc string. I was assuming you'd just `json.dumps(value).encode('utf-8')` to do the encoding and `json.loads(value.decode('utf-8'))` to pull the value back, but maybe you're trying to be more efficient than that. If so, should you perhaps error check the argument type, or, minimally, declare the argument type you expect (which has only informational and not runtime effect)?

File: pyproject.toml [hunk 2]
```diff
@@ -68,25 +69,28 @@ flaky = ">=3.7.0"
 pytest = ">=4.5.0"
 pytest-cov = ">=2.7.1"
 pytest-mock = ">=1.11.0"
+pytest-redis = "^2.0.0"
```
Comment (netsettler) [inline pyproject.toml:72]: I'll be curious what this contributes.

**Review by netsettler — APPROVED** [review]
Yeah, I don't see a blocker. Obviously, I've asked some design questions but if you're OK with your choices in the face of those questions, I think we can go with it.

**Review by dmichaels-harvard — APPROVED** [review]

**Review by dmichaels-harvard — APPROVED** [review]
LGTM

File: dcicutils/secrets_utils.py [hunk 3]
```diff
@@ -117,7 +117,8 @@ def apply_overrides(*, secrets: dict, rename_keys: Optional[dict] = None,
                 raise ValueError(f"Cannot rename {key} to {new_name}"
                                  f" because {a_or_an(new_name)} attribute already exists.")
             if key not in secrets:
-                raise ValueError(f"Cannot rename {key} to {new_name} because the secrets has no {key} attribute.")
+                logger.warning(f"Cannot rename {key} to {new_name} because the secrets has no {key} attribute.")
```
Comment (netsettler) [inline dcicutils/secrets_utils.py:120]: New situation needs new unit tests.

**Review by netsettler — APPROVED** [review]
My comment is not blocking.

File: test/test_redis_utils.py [hunk 4]
```diff
@@ -66,3 +68,23 @@ def build_item_metadata():
...
+        rd = RedisBase(redisdb)
+        my_key_meta = 'snovault:items:uuid:meta'
+        rd.set(my_key_meta, 'hello')
+        assert rd.get(my_key_meta) == 'hello'
+        rd.delete(my_key_meta)
+        # test exp with datetime
+        exp = datetime.timedelta(seconds=2)
+        rd.set(my_key_meta, 'hello', exp=exp)
+        assert rd.get(my_key_meta) == 'hello'
+        time.sleep(3)
```
Comment (netsettler) [inline test/test_redis_utils.py:83]: Is it possible to be using ControlledTime in this test to avoid having to have tests take a long time?

Comment (netsettler) [reply test/test_redis_utils.py:83]: I spoke to you on Slack, and you said not. Oh well.

**Review by netsettler — APPROVED** [review]
I already approved, but I've checked recent changes and still do.

File: dcicutils/redis_utils.py [hunk 5]
```diff
@@ -59,13 +60,17 @@ def info(self) -> dict:
...
         :param key: string to store value under
         :param value: value mapped to from key
+        :param exp: expiration time in seconds as int or datetime.timedelta (optional)
         :return: a "true" string <OK>
         """
-        return self.redis.set(self._encode_value(key), self._encode_value(value))
+        kwargs = {}
+        if exp:
+            kwargs['ex'] = exp
+        return self.redis.set(self._encode_value(key), self._encode_value(value), **kwargs)
```
Comment (dmichaels-harvard) [inline dcicutils/redis_utils.py:73]: Why using kwargs here instead of just `ex=exp` in redis.set call? (Nothing wrong with that just wondering).

Comment (willronchetti) [reply dcicutils/redis_utils.py:73]: Just future proofing it in case additional kwargs are needed in the future

**Pull request merged**